src/nvim/testdir/test_popup.vim

```diff
@@ -51,18 +51,6 @@ func Test_popup_complete()
   call assert_equal(["August"], getline(1,2))
   %d
 
-  " <BS> - Delete one character from the inserted text (state: 1)
-  " TODO: This should not end the completion, but it does.
-  " This should according to the documentation:
-  " January
-  " but instead, this does
-  " Januar
-  " (idea is, C-L inserts the match from the popup menu
-  " but if the menu is closed, it will insert the character <c-l>
-  call feedkeys("aJ\<f5>\<bs>\<c-l>\<esc>", 'tx')
-  call assert_equal(["Januar"], getline(1,2))
```

Comment:
instead of deleting the test, if we fixed the behavior we should test the new better behavior.